pkg/block/fetcher_test.go

```diff
@@ -393,3 +401,271 @@ func TestTimePartitionMetaFilter_Filter(t *testing.T) {
 	testutil.Equals(t, expected, input)
 
 }
+
+func TestCompactedBlockFilter_Filter(t *testing.T) {
+	for _, tcase := range []struct {
+		name     string
+		input    map[ulid.ULID]*metadata.Meta
+		expected func(input map[ulid.ULID]*metadata.Meta) map[ulid.ULID]*metadata.Meta
+	}{
+		{
+			name: "3 non compacted metas in bucket",
+			input: map[ulid.ULID]*metadata.Meta{
+				ULID(1): {
+					BlockMeta: tsdb.BlockMeta{
+						ULID: ULID(1),
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ULID(1)},
+						},
+					},
+				},
+				ULID(2): {
+					BlockMeta: tsdb.BlockMeta{
+						ULID: ULID(2),
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ULID(2)},
+						},
+					},
+				},
+				ULID(3): {
+					BlockMeta: tsdb.BlockMeta{
+						ULID: ULID(3),
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ULID(3)},
+						},
+					},
+				},
+			},
+			expected: func(input map[ulid.ULID]*metadata.Meta) map[ulid.ULID]*metadata.Meta {
```

Comment:
Why this complex signature for expected field? Why not just expect certain slice of ulid.ULID? (: